distribution/integration/test-common/integration-test-utils/src/main/java/org/wso2/carbon/integration/common/utils/CarbonCommandToolsUtil.java

```diff
@@ -0,0 +1,412 @@
+    }
+
+    /**
+     * This method is to get carbon home.
+     *
+     * @param context - AutomationContext
+     * @return - carbon home
+     * @throws CarbonToolsIntegrationTestException - Error while setup carbon home from carbon zip file
+     */
+    public static String getCarbonHome(AutomationContext context)
+            throws CarbonToolsIntegrationTestException {
+        try {
+            if (carbonHome != null) {
+                return carbonHome;
+            }
+            String carbonZip = System.getProperty(FrameworkConstants.SYSTEM_PROPERTY_CARBON_ZIP_LOCATION);
+            CarbonServerManager carbonServerManager = new CarbonServerManager(context);
+            carbonHome = carbonServerManager.setUpCarbonHome(carbonZip);
+            return carbonHome;
+        } catch (Exception ex) {
+            log.error("Error while shutting down the server using default credentials", ex);
+            throw new CarbonToolsIntegrationTestException("Error while shutting down the server using " +
+                                                          "default credentials ", ex);
+        }
+    }
+
+    /**
+     * This method is to shutdown carbon server
+     *
+     * @param portOffset        - port offset
+     * @param automationContext - AutomationContext
+     * @throws CarbonToolsIntegrationTestException - Error occurred while shutdown the server
+     */
+    public static void serverShutdown(int portOffset,
+                                      AutomationContext automationContext)
+            throws CarbonToolsIntegrationTestException {
+        long time = System.currentTimeMillis() + DEFAULT_START_STOP_WAIT_MS;
+        log.info("Shutting down server..");
+        boolean logOutSuccess = false;
+        try {
+            if (ClientConnectionUtil.isPortOpen(
+                    Integer.parseInt(ExtensionConstants.SERVER_DEFAULT_HTTPS_PORT))) {
+
+                int httpsPort = Integer.parseInt(FrameworkConstants.SERVER_DEFAULT_HTTPS_PORT) + portOffset;
+                String url = automationContext.getContextUrls().getBackEndUrl();
+                String backendURL = url.replaceAll("(:\\d+)", ":" + httpsPort);
+
+                ServerAdminClient serverAdminServiceClient =
+                        new ServerAdminClient(backendURL, automationContext.getContextTenant().
+                                getTenantAdmin().getUserName(), automationContext.getContextTenant().
+                                getTenantAdmin().getPassword());
+
+                serverAdminServiceClient.shutdown();
+                while (System.currentTimeMillis() < time && !logOutSuccess) {
+                    logOutSuccess = isServerDown(automationContext, portOffset);
+                    // wait until server shutdown is completed
+                }
+                log.info("Server stopped successfully...");
+            }
+        } catch (Exception ex) {
```

Comment:
serverAdminServiceClient.shutdown() throws Exception, at the moment i don't have control on it i discussed about this with the team, we will fix it in the next test framework release to send custom or specific exception properly.I have already replied this in pull request https://github.com/wso2/carbon4-kernel/pull/153